src/ppl/src/IOPlacer.cpp

```diff
@@ -933,6 +1174,40 @@ void IOPlacer::findSections(int begin,
   }
 }
 
+void IOPlacer::findSectionsPolygon(int begin,
+                                   int end,
+                                   odb::Line poly_edge,
+                                   std::vector<Section>& sections)
+{
+  int end_slot = 0;
+  while (end_slot < end) {
+    int blocked_slots = 0;
+    end_slot = begin + slots_per_section_ - 1;
+    if (end_slot > end) {
+      end_slot = end;
+    }
+    for (int i = begin; i <= end_slot; ++i) {
+      if (slots_[i].blocked) {
+        blocked_slots++;
+      }
+    }
+    int half_length_pt = begin + (end_slot - begin) / 2;
+    Section n_sec;
+    n_sec.pos = slots_.at(half_length_pt).pos;
+    n_sec.num_slots = end_slot - begin - blocked_slots + 1;
+    if (n_sec.num_slots < 0) {
+      logger_->error(PPL, 940, "Negative number of slots.");
+    }
+    n_sec.begin_slot = begin;
+    n_sec.end_slot = end_slot;
+    n_sec.used_slots = 0;
+    n_sec.edge = Edge::polygonEdge;
+
+    sections.push_back(n_sec);
+    begin = ++end_slot;
+  }
+}
```

Comment:
Notice how the `odb::Line poly_edge` is not even used inside the function. You should have only the `findSections` function, and pass a `bool` determine if the section will be for an `Edge::polygonEdge` or for a regular Edge.

Reply:
Now that I notice, this function is not even used in your code. This can just be removed.